vue/components/ui/molecules/bezier-curve/bezier-curve.vue

```diff
@@ -0,0 +1,377 @@
+        },
+        x2(value) {
+            this.$set(this.bezier, "x2", value);
+        },
+        y2(value) {
+            this.$set(this.bezier, "y2", value);
+        },
+        cx1(value) {
+            this.$set(this.bezier, "cx1", value);
+        },
+        cy1(value) {
+            this.$set(this.bezier, "cy1", value);
+        },
+        cx2(value) {
+            this.$set(this.bezier, "cx2", value);
+        },
+        cy2(value) {
+            this.$set(this.bezier, "cy2", value);
+        },
+        dragging(value) {
+            if (value && this.emitInterval !== null) this.startReactiveEmits();
+        }
+    },
+    computed: {
+        path() {
+            return `M ${this.bezier.x1}, ${this.bezier.y1} C ${this.bezier.cx1}, ${this.bezier.cy1}, ${this.bezier.cx2}, ${this.bezier.cy2}, ${this.bezier.x2}, ${this.bezier.y2}`;
+        },
+        viewBox() {
+            const [width, height] = this.resolution.split("x");
+            return `0, 0, ${width}, ${height}`;
+        },
+        classes() {
+            const base = {};
+            if (this.disabled) base.disabled = this.disabled;
+            return base;
+        }
+    },
+    methods: {
+        validCoordinates({ x, y }) {
+            const [maxWidth, maxHeight] = this.resolution.split("x").map(n => parseFloat(n));
+            if (x < 0 || y < 0 || x > maxWidth || y > maxHeight) return false;
+            return true;
+        },
+        /**
+         * @returns {Object} with coordinates label for a given point `target`.
+         */
+        getCoordinatesKey(target) {
+            switch (target) {
+                case "endpoint-1":
+                    return { x: "x1", y: "y1" };
+                case "endpoint-2":
+                    return { x: "x2", y: "y2" };
+                case "controlpoint-1":
+                    return { x: "cx1", y: "cy1" };
+                case "controlpoint-2":
+                    return { x: "cx2", y: "cy2" };
+                default:
+                    return { x: "x", y: "y" };
+            }
+        },
+        /**
+         * Gets the user cursor position considering the SVG matrix axis.
+         */
+        getCursorInSvgPos(event) {
+            const svgMatrix = this.$refs.svg.getScreenCTM();
+            const deviceEvent = event.type === "touchmove" ? event.touches[0] : event;
+            return {
+                x: (deviceEvent.clientX - svgMatrix.e) / svgMatrix.a,
+                y: (deviceEvent.clientY - svgMatrix.f) / svgMatrix.d
+            };
+        },
+        setPosition(point, position) {
+            const coordinatesKey = this.getCoordinatesKey(point);
+            this.$set(this.bezier, coordinatesKey.x, position.x);
+            this.$set(this.bezier, coordinatesKey.y, position.y);
+        },
+        /**
+         * Starts emitting the current drag once every `emitInterval` milliseconds.
+         */
+        startReactiveEmits() {

```

Comment:
Nothing reactive about this emit right? The jsdoc reflects that. Rename to startEmitInterval or something like that. Or just get rid of the function.